Jenkinsfile

```diff
@@ -70,15 +70,23 @@ pipeline {
     stage('Checkout') {
       options { skipDefaultCheckout() }
       steps {
-        pipelineManager([ cancelPreviousRunningBuilds: [ when: 'PR' ] ])
         deleteDir()
         // reference repo causes issues while running on Windows with the git-commit-id-maven-plugin
         gitCheckout(basedir: "${BASE_DIR}", githubNotifyFirstTimeContributor: true, shallow: false)
         stash allowEmpty: true, name: 'source', useDefaultExcludes: false
         script {
           dir("${BASE_DIR}"){
+            // Compare with the previous git commit if possible, otherwise the target branch or the git base commit.
+            def from = env.GIT_PREVIOUS_COMMIT?.trim() ? "HEAD~1"
```

Comment:
When pushing commits one by one, this will trigger a build for each commit, but not when we have more than one commit. I think here we need to have "the git commit of the previous build" if there is such.

For example, if I'm doing the following, will it work as expected ?
- create commit1 with code change
- create commit2 with doc change
- push them

From what I understand, the build will be triggered but will only detect a doc change.